test/runtests.jl

```diff
@@ -774,6 +774,50 @@ end
     @test sincospi(0.13im) == (sinpi(0.13im), cospi(0.13im))
 end
 
+# https://github.com/JuliaLang/julia/pull/37065
+@testset "muladd" begin
+    A23 = reshape(1:6, 2,3)
+    B34 = reshape(1:12, 3,4) .+ im
+    u2 = [10,20]
+    v3 = [3,5,7] .+ im
+    w4 = [11,13,17,19im]
+
+    @test muladd(A23, B34, 100) == A23 * B34 .+ 100
+    @test muladd(A23, B34, u2) == A23 * B34 .+ u2
+    @test muladd(A23, B34, w4') == A23 * B34 .+ w4'
+    @test_throws DimensionMismatch muladd(B34, A23, 1)
+    @test_throws DimensionMismatch muladd(A23, B34, ones(2,4,1))
+
+    @test muladd(A23, v3, 100) == A23 * v3 .+ 100
+    @test muladd(A23, v3, u2) == A23 * v3 .+ u2
+    @test muladd(A23, v3, im) isa Vector{Complex{Int}}
+    @test_throws DimensionMismatch muladd(A23, v3, ones(2,2))
+
+    @test muladd(v3', B34, 0) isa Adjoint
+    @test muladd(v3', B34, 2im) == v3' * B34 .+ 2im
+    @test muladd(v3', B34, w4') == v3' * B34 .+ w4'
+    @test_throws DimensionMismatch muladd(v3', B34, ones(1,4))
+
+    @test muladd(u2, v3', 0) isa Matrix
+    @test muladd(u2, v3', 99) == u2 * v3' .+ 99
+    @test muladd(u2, v3', A23) == u2 * v3' .+ A23
+    @test_throws DimensionMismatch muladd(u2, v3', ones(2,3,4))
+
+    @test muladd(u2', u2, 0) isa Number
+    @test muladd(v3', v3, im) == dot(v3,v3) + im
+    @test_throws DimensionMismatch muladd(v3', v3, [1])
+
+    vofm = [rand(1:9,2,2) for _ in 1:3]
+    Mofm = [rand(1:9,2,2) for _ in 1:3, _ in 1:3]
+    if VERSION >= v"1.5"
+        # Julia 1.4 gets vofm' * vofm wrong, gives a scalar
+        @test muladd(vofm', vofm, vofm[1]) == vofm' * vofm .+ vofm[1] # inner
+    end
```

Comment:
If it's just about the RHS, could work around that:
```suggestion
    if VERSION >= v"1.5"
        # Julia 1.4 gets vofm' * vofm wrong, gives a scalar
        @test muladd(vofm', vofm, vofm[1]) == vofm' * vofm .+ vofm[1] # inner
    else
        @test muladd(vofm', vofm, vofm[1]) == only(convert(Matrix, vofm') * vofm) .+ vofm[1] # inner
    end
```

Reply:
Yes that's probably better. 

PR is paused BTW as, after the one to Base got merged, we discovered that `muladd` wasn't only in use for scalars as I thought, so the methods being added for arrays ought to be more careful. Will update once its correction is in.